eve: replace old uis repository in example-solution builder

alexandre-allard opened the pull request:
**Component**: eve

**Context**:
Solution example cannot be built anymore in the CI

**Summary**:
UIS repository is used to install git2 on CentOS7 because otherwise the latest version available is 1.8.3.1 which is not compatible with the operator-sdk.

The repository has recently changed its URL, so we update it accordingly.
Plus, the package name has changed from git2u to git224.

We also add an option to yum install for it to fail when it cannot find a requested package, avoiding
ourselves to lost time on searching why the build is not working anymore (just because git was not in the good version).

**Acceptance criteria**:
We can build the Solution Example from the CI (in post-merge)

alexandre-allard committed:
eve: replace old uis repository in example-solution builder

UIS repository is used to install git2 on CentOS7
because otherwise the latest version available is 1.8.3.1
which is not compatible with the operator-sdk.

The repository has recently changed its URL, so we
update it accordingly.
Plus, the package name has changed from git2u to
git224.

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/replace-old-uis-repository-for-git` into
`development/2.4`
* [w/2.5/bugfix/replace-old-uis-repository-for-git](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/replace-old-uis-repository-for-git) will be merged into `development/2.5` 
* [w/2.6/bugfix/replace-old-uis-repository-for-git](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/bugfix/replace-old-uis-repository-for-git) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

bert-e commented:
# History mismatch



Merge commit #52832fac7139a240e859b45c692419a0c6b33e61 on the integration branch
`w/2.5/bugfix/replace-old-uis-repository-for-git` is merging a branch which is neither the current
branch `bugfix/replace-old-uis-repository-for-git` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `bugfix/replace-old-uis-repository-for-git` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/replace-old-uis-repository-for-git` into
`development/2.4`
* [w/2.5/bugfix/replace-old-uis-repository-for-git](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/replace-old-uis-repository-for-git) will be merged into `development/2.5` 
* [w/2.6/bugfix/replace-old-uis-repository-for-git](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/bugfix/replace-old-uis-repository-for-git) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

Ebaneck commented on eve/workers/pod-example-solution-builder/Dockerfile:
```diff
@@ -5,25 +5,25 @@ ARG OPERATOR_SDK_VERSION=v0.16.0
 WORKDIR /home/eve/workspace
 
-RUN yum install -y \
+RUN yum install -y --setopt=skip_missing_names_on_install=False \
```
Not sure we need this extra check here!
should `yum` not automatically fail if it cannot find any of the packages

Ebaneck replied:
Hmm, seems yum will not fail when it cannot find a package from the list provided so let's keep it.

alexandre-allard replied:
Yup, that's why I've added this option, the build would have failed here with a clear error message if we had this option set.

Ebaneck replied:
ok

alexandre-allard replied:
Well... not a clear message because it would have failed when provisioning the worker, so it returns nothing... but at least it's easily reproducible, just by building this Dockerfile.

Ebaneck commented on eve/workers/pod-example-solution-builder/Dockerfile:
```diff
@@ -5,25 +5,25 @@ ARG OPERATOR_SDK_VERSION=v0.16.0
     yum-utils \
+    yum-plugin-replace \
     docker-ce-cli-18.09.6 \
+    && yum replace -y git --replace-with git224 \
```
Seems the `IUS` repo does not have the `git` package at all...

And if we must pin a git version may we just use `git224-all`?
Source: https://repo.ius.io/7/x86_64/packages/g/

alexandre-allard replied:
No, git package comes from CentOS repositories.
That's basically what's done here.

Ebaneck reviewed (commented):
Few comments!

alexandre-allard committed:
eve: fix package name hardlinks to hardlink

The requested package was wrongly named `hardlinks`
instead of `hardlink`

alexandre-allard committed:
eve: set yum skip_missing_names_on_install=False opt

Add an option to yum install for it to fail
when it cannot find a requested package, avoiding
ourselves to lost time on searching why the build
is not working anymore (just because a package
could not be installed or is not in the good
version).

bert-e commented:
# History mismatch



Merge commit #fdfb0daf35f091cfd397e1c45d5427c658efb816 on the integration branch
`w/2.5/bugfix/replace-old-uis-repository-for-git` is merging a branch which is neither the current
branch `bugfix/replace-old-uis-repository-for-git` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `bugfix/replace-old-uis-repository-for-git` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/replace-old-uis-repository-for-git` into
`development/2.4`
* [w/2.5/bugfix/replace-old-uis-repository-for-git](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/replace-old-uis-repository-for-git) will be merged into `development/2.5` 
* [w/2.6/bugfix/replace-old-uis-repository-for-git](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/bugfix/replace-old-uis-repository-for-git) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

alexandre-allard commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality










*The following options are set:* **approve**

Ebaneck reviewed (approved):
Ship it!

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
